komodo-relational/src/main/java/org/komodo/relational/workspace/WorkspaceManager.java

```diff
@@ -145,6 +193,34 @@ public static WorkspaceManager getInstance( Repository repository ) throws KExce
     }
 
     /**
+     * @param repository
+     *        the repository whose workspace manager is being requested (cannot be <code>null</code>)
+     * @param transaction
+     *        the transaction containing the user name of the owner of this workspace manager
+     *        (if <code>null</code> then this manager is owner by the system user and has the workspace root as its path)
+     *
+     * @return the singleton instance for the given repository (never <code>null</code>)
+     * @throws KException
+     *         if there is an error obtaining the workspace manager
+     */
+    public static WorkspaceManager getInstance( Repository repository, UnitOfWork transaction) throws KException {
```

Comment:
True, but the transaction username is extracted and used so is purely a convenience method to avoid WorkspaceManager.getInstance(_repo, transaction.getUsername()) everywhere. getInstance(_repo, transaction) is so much tidier.